internal/processors/cloudeventconvert/cloudeventconvert.go

```diff
@@ -195,7 +196,25 @@ func (c *cloudeventProcessor) validateSignature(event *cloudevent.CloudEvent[jso
 		return false, fmt.Errorf("failed to unmarshal public key: %w", err)
 	}
 	recoveredAddress := crypto.PubkeyToAddress(*pubKey)
-	return common.HexToAddress(strings.TrimSpace(attestor)) == recoveredAddress, nil
+	if source != recoveredAddress {
+		return c.verifyERC1271Signature(signature, msgHash, source)
+	}
+
+	return true, nil
+}
+
+func (c *cloudeventProcessor) verifyERC1271Signature(signature []byte, msgHash []byte, source common.Address) (bool, error) {
+	contract, err := web3.NewErc1271(source, c.ethClient)
+	if err != nil {
+		return false, fmt.Errorf("failed to connect to address: %s: %w", source, err)
+	}
```

Comment:
do we need to do this every call or is it safe to call once in the constructor?

Reply:
There are ways around this, and I agree it's kind of heavy, but the problem with doing it in the constructor is that the contract address (here, `source`) is going to change between calls.

Reply:
I think we could revisit this by constructing our own thing to pass into `Call`.